File: Assignment 12/Question4.c
#include <stdio.h>

int FactorSumEven(int iNo)
{
    int iCnt = 0 , iSum = 0;
    for(iCnt = 1 ; iCnt<=(iNo/2) ; iCnt++)
    {
        if(iNo % iCnt ==0)
        {
            if(iCnt % 2 ==0 )
            {
            iSum = iSum +iCnt;  
            }    
        }
    }
    return iSum ;
}

int main()
{
    int iValue = 0 ,iRet = 0;


    printf("Enter Number : ");
    scanf("%d",&iValue);

    iRet = FactorSumEven(iValue);

    printf("%d is the sum of Even factors",iRet);

    return 0;
}

File: Assignment 21/Question4.java
import java.util.Scanner;

class Logic
{
    
    public int TotalFactor(int iNo) 
    {
        int iCnt = 0;   
        int iCount = 0 ;
        for (iCnt = 1 ; iCnt<=(iNo/2) ; iCnt++)
        {
            if (iNo % iCnt == 0)
            {
                iCount = iCount +1;
            }
        }
        return iCount;
       
    }  
}          
class Question4
{
    public static void main(String[] args) 
    {
        
        Scanner sobj = new Scanner(System.in);
        int iValue1 = 0   ,iRet = 0;
       
        System.out.println("Enter your Number :");
        iValue1 = sobj.nextInt();

        Logic lobj = new Logic();

        iRet = lobj.TotalFactor(iValue1);

        System.out.println(" The Total No of Factors are : "+iRet);
      
    }    
}
